$NetBSD: patch-configure.ac,v 1.3 2026/10/29 16:28:04 gutteridge Exp $

Modernize C tests to fix builds with GCC >= 14.
https://github.com/SawfishWM/librep/commit/48f557ab34d47a7a1fd9e8425542f720be40946e

Account for loongarch64 and riscv stack direction.
https://github.com/SawfishWM/librep/commit/e668e25fd994a76275d788e16f290931372773b5
https://github.com/SawfishWM/librep/commit/52d64573cf4e3e6d92ef7ec2b72a73abdccd0389

--- configure.ac.orig	2017-08-25 19:59:05.000000000 +0000
+++ configure.ac
@@ -432,15 +432,18 @@ if test "${with_value_type}" = "undef"; then
 			   as a pointer. [TYPE=<probed>] (see README)], [],
  [with_value_type="undef"])
 if test "${with_value_type}" = "undef"; then
-  AC_TRY_RUN([main () { exit (!(sizeof (int) >= sizeof (void *)));}],
+  AC_TRY_RUN([#include <stdlib.h>
+	       int main () { exit (!(sizeof (int) >= sizeof (void *)));}],
 	     [with_value_type=int])
 fi
 if test "${with_value_type}" = "undef"; then
-  AC_TRY_RUN([main () { exit (!(sizeof (long int) >= sizeof (void *)));}],
+  AC_TRY_RUN([#include <stdlib.h>
+	       int main () { exit (!(sizeof (long int) >= sizeof (void *)));}],
 	     [with_value_type="long int"])
 fi
 if test "${with_value_type}" = "undef"; then
-  AC_TRY_RUN([main () { exit (!(sizeof (long long int) >= sizeof (void *)));}],
+  AC_TRY_RUN([#include <stdlib.h>
+	       int main () { exit (!(sizeof (long long int) >= sizeof (void *)));}],
 	     [with_value_type="long long int"])
 fi
 if test "${with_value_type}" = "undef"; then
@@ -457,7 +460,8 @@ if test "${with_value_sizeof}" = "undef"; then
 if test "${with_value_sizeof}" = "undef"; then
   dnl the following fragment is inspired by AC_CHECK_SIZEOF
   AC_TRY_RUN([#include <stdio.h>
-	      main () {
+	      #include <stdlib.h>
+	      int main () {
 		FILE *f = fopen ("conftestval", "w");
 		if (!f) exit (1);
 		fprintf (f, "%d\n", sizeof (${with_value_type}));
@@ -536,7 +540,7 @@ if test "${with_stack_direction}" = unknown; then
  [with_stack_direction=unknown])
 if test "${with_stack_direction}" = unknown; then
   case ${host_cpu} in
-    sparc*|i?86*|powerpc*|x86_64*|arm*|aarch64*|s390*|mips*|alpha*|ia64*)
+    sparc*|i?86*|powerpc*|x86_64*|arm*|aarch64*|s390*|mips*|alpha*|ia64*|loongarch64*|riscv*)
       AC_MSG_RESULT([assuming downwards])
       with_stack_direction="-1"
       ;;
@@ -548,7 +552,8 @@ if test "${with_stack_direction}" = unknown; then
   esac
 fi
 if test "${with_stack_direction}" = unknown; then
-  AC_TRY_RUN([ int level = 1;
+  AC_TRY_RUN([ #include <stdlib.h>
+	       int level = 1;
 	       void inner (char *foo) { char bar;
 		 if (level) { --level; inner (foo); }
 		 exit (!(foo >= &bar)); }
